libvirt/cloudinit_def.go

```diff
@@ -197,11 +198,16 @@ func (ci *defCloudInit) createFiles() (string, error) {
 		return "", fmt.Errorf("Error while writing user-data to file: %s", err)
 	}
 
-	metadata, err := yaml.Marshal(&ci.MetaData)
+	mergedMetaData, err := mergeMetaDataIntoMetaDataRaw(ci.MetaData, ci.MetaDataRaw)
 	if err != nil {
-		return "", fmt.Errorf("Error dumping cloudinit's meta data: %s", err)
+		return "", fmt.Errorf("Error merging Metadata with MetadataRaw: %v", err)
 	}
-	if err = ioutil.WriteFile(filepath.Join(tmpDir, metaData), metadata, os.ModePerm); err != nil {
+	metadata := fmt.Sprintf("%s", mergedMetaData)
```

Comment:
Maybe I didn't have enough coffee yet this morning, but do we actually need to do that? `mergedMetaData` is already a `string`.